server/user-groups.ts

```diff
@@ -213,11 +213,14 @@ export class RoomAuth extends Auth {
 		super();
 		this.room = room;
 	}
-	get(user: ID | User): GroupSymbol {
+	get(userOrID: ID | User): GroupSymbol {
+		const user = typeof userOrID === 'string' ? null : userOrID as User;
+		const id = user ? user.id : userOrID as ID;
+		if (user && !user.registered) return Auth.defaultSymbol();
```

Comment:
This means that unregistered users won't get player rank in battles, is this intended?

Reply:
Ah, no, that's not intended. But it may be preferable?

Reply:
We could hardcode an exception for temporary rooms.

Reply:
(I ended up replacing this with a trusted user check, which no longer has this problem.)